src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3107,14 +3111,47 @@ protected static unsafe void UseAllDocumentsEtag(DocumentsOperationContext docum
                 //buffer[2] - last processed doc etag
                 //buffer[3] - last process tombstone etag
             }
-        }
 
-        protected int MinimumSizeForCalculateIndexEtagLength()
+            if (q.CounterIncludes != null || q.HasCounterSelect)
+            {
+                Debug.Assert(length > sizeof(long) * 5);
```

Comment:
Add a debug message here, if not we will not know during test execution where it failed.